openapi3filter/req_resp_decoder.go

```diff
@@ -224,20 +224,65 @@ func decodeStyledParameter(param *openapi3.Parameter, input *RequestValidationIn
 
 func decodeValue(dec valueDecoder, param string, sm *openapi3.SerializationMethod, schema *openapi3.SchemaRef) (interface{}, error) {
 	var decodeFn func(param string, sm *openapi3.SerializationMethod, schema *openapi3.SchemaRef) (interface{}, error)
-	switch schema.Value.Type {
-	case "array":
-		decodeFn = func(param string, sm *openapi3.SerializationMethod, schema *openapi3.SchemaRef) (interface{}, error) {
-			return dec.DecodeArray(param, sm, schema)
+	if len(schema.Value.AnyOf) > 0 {
+		for _, sr := range schema.Value.AnyOf {
+			value, err := decodeValue(dec, param, sm, sr)
+			if value == nil && err == nil {
+				continue
+			}
+			return value, err
+		}
+		return nil, nil
+	}
+	if len(schema.Value.AllOf) > 0 {
+		isMatched := true
+		var value interface{}
+		var err error
+		for _, sr := range schema.Value.AllOf {
+			value, err = decodeValue(dec, param, sm, sr)
+			if value == nil {
```

Comment:
Shouldn't this be checking `err` instead, as `value` can be `nil` for `type: null` values?

Reply:
decodeValue function will return nil,nil if type is null. here judge "value==nil" means the value doesn't match this schema rule,and no longer to match other schema rules. Yes ,maybe here have a bug. when err isn't nil.

Reply:
Could checking `err` before `value` help?

Reply:
I have updated this.